commands/command_migrate_import.go

```diff
@@ -202,3 +283,98 @@ func trackedToBlob(db *odb.ObjectDatabase, patterns *tools.OrderedSet) ([]byte,
 		Size:     int64(attrs.Len()),
 	})
 }
+
+// rewriteTree replaces the blob at the provided path within the given tree with
+// a git lfs pointer. It will recursively rewrite any subtrees along the path to the
+// blob.
+func rewriteTree(db *odb.ObjectDatabase, root []byte, path string) ([]byte, error) {
+	tree, err := db.Tree(root)
+	if err != nil {
+		return nil, err
+	}
+
+	splits := strings.SplitN(path, "/", 2)
+
+	switch len(splits) {
+	case 1:
+		// The path points to an entry at the root of this tree, so it must be a blob.
+		// Try to replace this blob with a git lfs pointer.
+		index := findEntry(tree, splits[0])
+		if index < 0 {
+			return nil, errors.Errorf("unable to find entry %s in tree", splits[0])
+		}
+
+		blobEntry := tree.Entries[index]
+		blob, err := db.Blob(blobEntry.Oid)
+		if err != nil {
+			return nil, err
+		}
+
+		var buf bytes.Buffer
+
+		if _, err := clean(lfs.NewGitFilter(cfg), &buf, blob.Contents, blobEntry.Name, blob.Size); err != nil {
```

Comment:
Why do we need to pass `lfs.NewGitFilter(cfg)`? I can't recall how the current implementation of `clean()` works, and whether or not we need this. If we do need it, can it be saved elsewhere instead of being constructed every time?

Reply:
It looks like it's being used in a few places in the `clean()` implementation, but it shouldn't change throughout the course of the entire operation. I agree that it makes more sense to construct one in `migrateImportCommand()` and pass it down into `rewriteTree()`, so I'll go ahead and change that